packages/react/src/hooks/use-last-factor.ts

```diff
@@ -0,0 +1,73 @@
+import { useEffect, useMemo, useCallback } from "react";
+import { isBrowser } from "@slashid/react-primitives";
+import { Handle, isFactor } from "../domain/types";
+import { useConfiguration } from "./use-configuration";
+import { useSlashID } from "./use-slash-id";
+import { Factor } from "@slashid/slashid";
+
+export const STORAGE_LAST_FACTOR_KEY = "@slashid/LAST_FACTOR";
+
+type UseLastFactor = () => {
+  lastFactor: Factor | undefined;
+};
+
+type SuccessEvent = {
+  authenticationFactor: Handle | undefined;
+};
+
+export const useLastFactor: UseLastFactor = () => {
+  const { storeLastFactor } = useConfiguration();
+  const { sid } = useSlashID();
+
+  const lastFactor = useMemo(() => {
+    if (!isBrowser()) {
+      return undefined;
+    }
+
+    try {
+      const storedFactor = window.localStorage.getItem(STORAGE_LAST_FACTOR_KEY);
```

Comment:
Don't we have an abstraction for storage? 🤔

Reply:
not really, we have a helper to create a storage for `tokenStorage`, other than that we're using local storage if possible